### DISCUSSION TITLE
Add `gd` extension

### DISCUSSION THREAD

**Discussion by Maxoplata** [NativePHP/laravel#281] [Ideas]
I'm assuming this is the correct repo for this issue, but I'm not 100% sure. 

I am able to use `imagecreatefromstring()` to load PNG and GIF images, but not JPEG images. When I try to load a JPEG, I get this error:

> imagecreatefromstring(): No JPEG support in this PHP build

Here is a code example that causes the error:

```
$fileData = file_get_contents('/path/to/image.jpg');
$image = imagecreatefromstring($fileData);
```

**Comment by vitorleonel** [comment]
I saw that we aren't compiling the PHP with the `gd` extension. It should be the case.

@mpociot could it be included? I mean, I saw some PR saying that we are keeping a small list of extensions, so that's why I'm asking.

Are you able to compile the PHP with that extension (gd) @Maxoplata? So you can validate it and we could know that this is the case.

**Comment by simonhamp** [comment]
The specific extensions we can add will depend upon whether they're ready for cross-platform support.

Please consult the [static-php-cli Extensions table](https://static-php.dev/en/guide/extensions.html) for details of which extensions are available across the major platforms.

As of this writing, `gd` is not available for static builds for Windows. [I've requested it](https://github.com/crazywhalecc/static-php-cli/issues/429). When it is, then I'll look to add it to future builds 👍🏼

Will leave this issue open until then.

**Reply by aftabkhaliq86** [reply]
@simonhamp  CAN WE HAVE EXTENSION sockets because it is supported and available across the major platform

**Reply by simonhamp** [reply]
@aftabkhaliq86 The `sockets` extension is already included

**Comment by CoolDude53** [comment]
@simonhamp looks like gd is good to go now...

https://github.com/crazywhalecc/static-php-cli/pull/484

**Reply by simonhamp** [reply]
Awesome! I'll get it added over the coming days.

Static building for the latest versions of PHP 8.2 and 8.3 [seems to be broken](https://github.com/crazywhalecc/static-php-cli/issues/527) right now, but once this is resolved we'll get the latest versions up and running with GD

**Comment by simonhamp** [comment]
Thi has been added as of [`php-bin` v0.5.0](https://github.com/NativePHP/php-bin/releases/tag/0.5.0)
